qa/common/common.sh

```diff
@@ -577,3 +570,42 @@ echo "Result: OK"
 EOF
     _run_test_script_on_node $TESTSCRIPT $RGWNODE
 }
+
+function test_systemd_ceph_osd_target_wants {
+  #
+  # see bsc#1051598 in which ceph-disk was omitting --runtime when it enabled
+  # ceph-osd@$ID.service units
+  #
+  local TESTSCRIPT=/tmp/test_systemd_ceph_osd_target_wants.sh
+  local STORAGENODE=$(_first_x_node storage)
+  cat << 'EOF' > $TESTSCRIPT
+set -ex
+trap 'echo "Result: NOT_OK"' ERR
+local CEPH_OSD_WANTS="/systemd/system/ceph-osd.target.wants"
+local ETC_CEPH_OSD_WANTS="/etc$CEPH_OSD_WANTS"
+local RUN_CEPH_OSD_WANTS="/run$CEPH_OSD_WANTS"
+local DIR_CONTENTS
+set +e
+ls -l $ETC_CEPH_OSD_WANTS
+ls -l $RUN_CEPH_OSD_WANTS
+# if $ETC_... is not a directory and $RUN_... is, consider the test a pass
+if [ -d "$RUN_CEPH_OSD_WANTS" -a ! -d "$ETC_CEPH_OSD_WANTS" ] ; then
+    echo "ceph-osd.target symlinks appear to be in the right place"
+    return 0
+fi
+set -e
+# if $ETC_... is a directory, it should at least be empty
+if [ -d "$ETC_CEPH_OSD_WANTS" ] ; then
+    DIR_CONTENTS=$(ls --almost-all $ETC_CEPH_OSD_WANTS)
+    echo "$ETC_CEPH_OSD_WANTS exists and is a directory: asserting that it is empty"
+    test -z "$DIR_CONTENTS"
+fi
```

Comment:
@tserong So you think we could get away with a mere `! test -d "$ETC_CEPH_OSD_WANTS"` instead of this if clause?

Reply:
I think so (I'm almost certain I've never seen those directories exist and be empty -- presumably that would only happen if someone was manually screwing around with `mkdir` and whatnot, rather than using `systemctl`)